src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -372,7 +376,7 @@ protected function createNode()
                 $node->setKeyAttribute($this->key, $this->removeKeyItem);
             }
 
-            if (true === $this->atLeastOne) {
+            if (true === $this->atLeastOne || false === $this->allowEmptyValue) {
```

Comment:
or given this bug, shouldn't we *not* fix it (to prevent any bc break), and deprecate `cannotBeEmpty` instead?